.travis.yml

```diff
@@ -0,0 +1,77 @@
+language: node_js
+node_js:
+  - 18.15.0
+
+services:
+  - docker
+
+branches:
+  only:
+    - qa
+    - production
+
+jobs:
+  include:
+    - stage: docker build
+      if: (branch = qa or branch = production)
+      before_install:
+        - if [[ $TRAVIS_BRANCH == "qa" ]];
+          then
+          export AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID_QA};
+          export AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY_QA};
+          export ENV="qa";
+          export ECR_URL=491147561046.dkr.ecr.us-east-1.amazonaws.com/nyplglobalheader;
+          fi
+        - if [[ "$TRAVIS_BRANCH" == "production" ]];
+          then
+          export AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID_PRODUCTION};
+          export AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY_PRODUCTION};
+          export ENV="production";
+          export ECR_URL=463786892988.dkr.ecr.us-east-1.amazonaws.com/nyplglobalheader;
+          fi
+      env:
+        - AWS_DEFAULT_REGION=us-east-1
+        - AWS_REGION=us-east-1
+      install:
+        - pip install --user awscli
+        - export LOCAL_TAG_NAME="${ENV}-latest"
+      script:
+        - eval $(aws ecr get-login --no-include-email --region us-east-1)
+        - DOCKER_BUILDKIT=1 docker build -t "$LOCAL_TAG_NAME" --secret id=AWS_ACCESS_KEY_ID --secret id=AWS_SECRET_ACCESS_KEY" .
+        - docker tag "$LOCAL_TAG_NAME" "$ECR_URL:$LOCAL_TAG_NAME"
+        - docker push "$ECR_URL:$LOCAL_TAG_NAME"
+
+    - stage: deploy qa
+      if: branch IN (qa) AND type != pull_request
+      env:
+        - AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID_QA
```

Comment:
Just curious about the syntax: this line doesn't have the var value wrapped with braces as `${...}` but on line 20 it is wrapped in braces. Typo or should both work?

Reply:
Both should work fine, I removed the braces for consistency.